addons/sourcemod/scripting/gokz-hud/tp_menu.sp

```diff
@@ -180,7 +217,7 @@ static void TPMenuAddItemTeleport(KZPlayer player, Menu menu)
 		Format(display, sizeof(display), "%s #%d", display, player.TeleportCount);
 	}
 	
-	if (player.CanTeleportToCheckpoint)
+	if (player.CanTeleportToCheckpoint || !gB_DynamicMenu[player.ID])
```

Comment:
Why does having the dynamic menu disabled bypass all of those checks? Wouldn't that mean that those menu items will never grey out without a dynamic menu? I believe only the menu items that weren't behind a conditional need the added dynamic menu condition.

Reply:
> Why does having the dynamic menu disabled bypass all of those checks? Wouldn't that mean that those menu items will never grey out without a dynamic menu? 
Yes. I don't think that's a problem though, it helps players with high ping, where they can quickly set a checkpoint then teleport without having to wait from the server to confirm that there is at least one checkpoint placed.

> I believe only the menu items that weren't behind a conditional need the added dynamic menu condition.
This doesn't make too much sense to me, it's dynamic because the condition changes over time.

Reply:
The main issue I see is that it changes the current behavior where only a part of the menu is dynamic and there would be no way to get that back with the change, so it's all or nothing. I also can't really think of a situation where being able to TP instantly after setting the first CP would be useful with regards to ping.

Reply:
> The main issue I see is that it changes the current behavior where only a part of the menu is dynamic and there would be no way to get that back with the change, so it's all or nothing.

I could make it an option choice that preserves the old behavior as well, if you think that is necessary.

> I also can't really think of a situation where being able to TP instantly after setting the first CP would be useful with regards to ping.

On high ping, I accidentally walked off the block quite often, and if I have no checkpoint before that I would try to press 2 immediately but nothing would happen. It is also useful for resetting momentum on those very rare instances, though probably only in VNL because other modes have way faster deceleration.

Reply:
I guess I'd make an option for it, then. I'm a bit wary of option bloat, but since we add an option for it already anyway we might as well make it a 3-way toggle.